### PR TITLE
no user can delete any resource

### LINKED ISSUE #801: Ensure no users can delete items
Our UI does not include delete buttons for any types of resource (things can only be made not active).

But, our permissions table doesn't explicitly deny power-users from being able to delete things.  We should explicitly state that in the data, to ensure that in future maintenance a `delete` button isn't accidentally left enabled

image

### PR DESCRIPTION
**Pull request by TahaKhanAbdalli**
fixes #801

### CODE REVIEW

**Comment by IanMayo** [comment]
Thanks for this @TahaKhanAbdalli , can you remind me how we implement permissions for users without a role?  (so, not `rco-user` or `rco-power-user`).

File: src/providers/authProvider/permissions.ts [hunk 1]
```diff
@@ -1,22 +1,27 @@
 import * as constants from '../../constants'
 type UserRoles = UserRole
```
Comment (IanMayo) [inline src/providers/authProvider/permissions.ts:2]: is there a reason why we are renaming the `UserRole` type here @TahaKhanAbdalli?

**Review by IanMayo — CHANGES_REQUESTED** [review]
@TahaKhanAbdalli  - I'm in as `power-user` but I don't have `edit` permission for the `Users` table.

**Comment by TahaKhanAbdalli** [comment]
> Thanks for this @TahaKhanAbdalli , can you remind me how we implement permissions for users without a role? (so, not `rco-user` or `rco-power-user`).

I think we removed that because of the requirements.

**Comment by TahaKhanAbdalli** [comment]
> @TahaKhanAbdalli - I'm in as `power-user` but I don't have `edit` permission for the `Users` table.
> 
> image

simple `rco-user` also has write permission to users. Do I remove that too?

**Comment by IanMayo** [comment]
> > Thanks for this @TahaKhanAbdalli , can you remind me how we implement permissions for users without a role? (so, not `rco-user` or `rco-power-user`).
> 
> I think we removed that because of the requirements change.

Aah yes, we did that here:
https://github.com/DeepBlueCLtd/RCO/commit/eaea7fe97ed2180eb4ccd1feafc1c61ad68a49d0#diff-fbd3282a67d2fc43a6f851ad4c86884a61d677bc927eabb2abfe551148ee760e

**Comment by IanMayo** [comment]
> > @TahaKhanAbdalli - I'm in as `power-user` but I don't have `edit` permission for the `Users` table.
> > image
> 
> simple `rco-user` also has write permission to users. Do I remove that too?

No, the issue is that my role was `missing` the ability to edit a user.  I'm ok with plain `rco-user` having write permission.

**Review by IanMayo — APPROVED** [review]

**Pull request merged**